provider/oauth2/forms.py

```diff
@@ -336,3 +336,10 @@ def clean(self):
 
         data['client'] = client
         return data
+
+
+class ClientCredentialsGrantForm(ScopeMixin, OAuthForm):
+    """
+    Validate a client credentials grant request.
+    """
+    scope = ScopeChoiceField(choices=SCOPE_NAMES, required=False)
```

Comment:
If you are supporting scope, then I believe you need to verify the scope checks out in this form's `clean` method.  For example: https://github.com/edx/django-oauth2-provider/blob/edx/provider/oauth2/forms.py#L260-L272